genie-agent/src/main/java/com/netflix/genie/agent/execution/locks/Lock.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.agent.execution.locks;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * A simple interface representing a closeable lock.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+public interface Lock extends Closeable {
+
+    /**
+     * Acquire a lock.
+     * @throws IOException in case of problem acquiring the lock
+     */
+    void lock() throws IOException;
```

Comment:
For a generic interface like this throwing just an `IOException` is relatively limiting if other implementations end up throwing other types of exceptions now you have to go back and modify this interface. Then all the implementations up till that point will break. Unless you're planning on forcing every other implementation to wrap their exceptions in an `IOException`. For this we've generally wrapped it in `GenieException` or some extension of `GenieException` specific to this like `GenieLockException`.

Reply:
Not intentional. Will use GenieException

Reply:
This is client side.. Created a LockException